backend/services/useradm/utils/similar_chcker.go

```diff
@@ -0,0 +1,30 @@
+package utils
+
+import (
+	"strings"
+
+	"github.com/mendersoftware/mender-server/services/useradm/model"
+)
+
+func CheckIfPassSimilarToEmailRaw(mail string, pass string) bool {
+	if len(mail) == 0 || len(pass) == 0 {
+		return false
+	}
+	lmail := strings.ToLower(mail)
+	lpass := strings.ToLower(pass)
+	return strings.Contains(lmail, lpass)
+}
+func CheckIfPassSimilarToEmail(user *model.User, userUpdate *model.UserUpdate) bool {
+	if len(userUpdate.Password) > 0 {
+		// check when changing the password
+		mail := string(user.Email)
+		if len(mail) == 0 {
+			mail = string(userUpdate.Email)
+		}
```

Comment:
This logic should be the opposite. You want to compare with the email in the state after performing the update.
```suggestion
		mail := string(userUpdate.Email)
		if len(mail) == 0 {
			mail = string(user.Email)
		}
```
Even better, you could compare with both the previous and the updated (if any).
```suggestion
		pwd := userUpdate.Password
		// Check if similar to updated email
		if userUpdate.Email != "" && CheckIfPassSimilarToEmailRaw(userUpdate.Email, pwd) {
			return true
		}
		// Check if similar to (old) email
		return CheckIfPassSimilarToEmailRaw(user.Email, pwd)
```